overwatch-rs/src/services/mod.rs

```diff
@@ -40,13 +40,12 @@ pub trait ServiceData {
 }
 
 /// Main trait for Services initialization and main loop hook
-#[async_trait]
 pub trait ServiceCore: Sized + ServiceData {
     /// Initialize the service with the given state
     fn init(service_state: ServiceStateHandle<Self>) -> Result<Self, super::DynError>;
 
     /// Service main loop
-    async fn run(mut self) -> Result<(), super::DynError>;
+    fn run(self) -> impl Future<Output = Result<(), super::DynError>> + Send;
```

Comment:
So, for async methods that return something that need to be send we need to use `impl Future` instead of `async`?

Reply:
Yes, if we use `async`, by default there is no `Send` bound.